src/components/Pagination/Pagination.tsx

```diff
@@ -2,0 +1,69 @@
+import React from 'react';
+
+type Props = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: React.Dispatch<React.SetStateAction<number>>;
+};
+
+export const Pagination = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}: Props) => {
+  const numOfPages = Math.ceil(total / perPage);
+  const pages = Array.from({ length: numOfPages }, (_, i) => i + 1);
+  const prevHandler = () => currentPage !== 0 && onPageChange(currentPage - 1);
+  const nextHandler = () =>
+    currentPage !== numOfPages - 1 && onPageChange(currentPage + 1);
+
+  return (
+    <>
+      <ul className="pagination">
+        <li className={`page-item ${currentPage === 0 && 'disabled'}`}>
+          <a
+            data-cy="prevLink"
+            className="page-link"
+            href="#prev"
+            aria-disabled={currentPage === 0 ? 'true' : 'false'}
+            onClick={prevHandler}
+          >
+            «
+          </a>
+        </li>
+        {pages.map((num: number) => {
+          return (
+            <li
+              className={`page-item ${num - 1 === currentPage && 'active'}`}
+              key={num}
+            >
+              <a
+                data-cy="pageLink"
+                className="page-link"
+                href={`#${num}`}
+                onClick={() => onPageChange(num - 1)}
+              >
+                {num}
+              </a>
+            </li>
+          );
+        })}
```

Comment:
You are creating a new function inside the map method. This can lead to performance issues because a new function is created for each item in the array on every render. You could improve this by defining a function outside of the map method and calling it inside the map method.